frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -188,6 +196,12 @@ export class Gallery extends React.Component {
     this.setState({ imagePreviewList, displayedPictureIndex })
   }
 
+  updateBreadcrumbsAndTitle = () => {
+    const { state, props } = this
+    this.buildBreadcrumbs(state.workspaceLabel, state.folderDetail, false)
```

Comment:
There might be an issue in this function.

the setState function of react is asynchronous.
The advantages of that is that React can take upon himself to batch potential multiple setState calls into one to avoid too many calls to `render ()`.

In your case, you are not guaranteed that the state has been updated by `handleWorkspaceModified` when executing `updateBreadcrumbsAndTitle`.
Therefore, the `this.buildBreadcrumbs(state.workspaceLabel, state.folderDetail, false)` might be done with the old workspace label.

The solution to that it to make `updateBreadcrumbsAndTitle` pure and calling it from `handleWorkspaceModified` with the data from the TLM as parameters